framework/src/operators/mod.rs

```diff
@@ -160,6 +172,33 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn filter_map_operator() {
+        use packets::icmp::v4::tests::ICMPV4_PACKET;
+        use packets::udp::tests::UDP_PACKET;
+
+        dpdk_test! {
+            let (producer, batch) = single_threaded_batch::<RawPacket>(2);
+            let mut batch = batch.filter_map(|p| {
+                let v4 = p.parse::<Ethernet>()?.parse::<Ipv4>()?;
+                if v4.protocol() == ProtocolNumbers::Udp {
+                    let mut eth  = v4.deparse();
+                    eth.swap_addresses();
+                    Ok(Some(eth.parse::<Ipv4>()?))
+                } else {
+                    Ok(None)
+                }
+            });
+
+            producer.enqueue(RawPacket::from_bytes(&UDP_PACKET).unwrap());
+            producer.enqueue(RawPacket::from_bytes(&ICMPV4_PACKET).unwrap());
+
+            let p1 = batch.next().unwrap().unwrap();
+            assert_eq!(ProtocolNumbers::Udp, p1.protocol());
```

Comment:
should assert something you've changed in the `filter_map`.

Reply:
ok